internal/ent/hooks/event.go

```diff
@@ -220,6 +228,38 @@ func RegisterListeners(e *Eventer) error {
 		return err
 	}
 
+	_, err = e.Emitter.On(OrganizationDelete, handleOrganizationDelete)
+	if err != nil {
+		log.Error().Err(ErrFailedToRegisterListener)
+		return err
+	}
+
+	_, err = e.Emitter.On(OrganizationDeleteOne, handleOrganizationDelete)
+	if err != nil {
+		log.Error().Err(ErrFailedToRegisterListener)
+		return err
+	}
+
+	return nil
+}
+
+// handleOrganizationDelete handles the deletion of an organization and deletes the customer in Stripe
+func handleOrganizationDelete(event soiree.Event) error {
+	client := event.Client().(*entgen.Client)
+	entMgr := client.EntitlementManager
+
+	if entMgr == nil {
+		log.Debug().Msg("EntitlementManager not found on client, skipping customer deletion")
+
+		return nil
+	}
+
+	if err := entMgr.FindAndDeactivateCustomerSubscription(event.Context(), lo.ValueOr(event.Properties(), "ID", "").(string)); err != nil {
+		log.Err(err).Msg("Failed to deactivate customer subscription")
```

Comment:
Same here:, also missing the level here

```suggestion
		zerolog.Ctx(event.Context()).Error().Err(err).Msg("Failed to deactivate customer subscription")
```